### ISSUE TITLE
How to read a avi file with VideoCaptureFile?

### ISSUE THREAD

**Issue by akihirom2017** [hybridgroup/gocv#238]
I used python + openCV before, but now I am trying to open an avi file with gocv.

## Description

I think VideoCapture is equivalent to gocv.VideoCaptureFile to open the file.
I simply try to read/open avi file with VideoCaptureFile.
however, this gave me an error message.
Do you have any hints/idea which I should take a look to fix this problem?

Thanks

## Steps to Reproduce

1. Run following command gave me an error message:
package main

```
import (
       "gocv.io/x/gocv"
        "fmt"
)

func main() {
  avi, _ := gocv.VideoCaptureFile("test.type11-103.lightreduce2.avi")
  fmt.Println(avi)
}
```
this resulted in:
Unable to stop the stream: Inappropriate ioctl for device
&{0x2053d10}

however, equivalent python code works without the same problem:
```
import cv2
import sys

cap = cv2.VideoCapture('test.type11-103.lightreduce2.avi')
print (cap)
#more functions to read frame by frame and save the frame etc...
```

## Your Environment

* Operating System and version: Ubuntu 16.04
* OpenCV version used: 3.4
* How did you install OpenCV? follow gocv instraction
* GoCV version used: current
* Go version: go1.8.1
* Did you run the `env.sh` or `env.cmd` script before trying to `go run` or `go build`?
Neither.

**Comment by deadprogram** [comment]
Hello @akihirom2017 in the `dev` branch you can run most of the cmd examples such as https://github.com/hybridgroup/gocv/blob/master/cmd/capwindow/main.go with a file input.

The following command plays back an AVI file in a window:

```shell
$ go run ./cmd/capwindow/main.go /home/ron/Downloads/vtest.avi
Start reading camera device: /home/ron/Downloads/vtest.avi
```

When the end of file is reached, the following is displayed by the above cmd example:

```shell
Error cannot read device /home/ron/Downloads/vtest.avi
```
Also you might want to update to Golang 1.9.x or 1.10.x which are the versions we officially support.

Hope that helps!

**Comment by deadprogram** [comment]
Glad to have been able to help.

Now closing, please reopen if needed.

**Comment by akihirom2017** [comment]
Sorry, I have got an error on the above setting.

I installed gocv-dev (& rename it gocv) and start using "./cmd/capwindow/main.go" as you mentioned above. in the suggested command line, I changed "/home/ron/Downloads/vtest.avi" to "./images/small.mp4" which exist in image directory or avi files I have .
  
Then, I I have got a following error message:
Unable to stop the stream: Inappropriate ioctl for device

As far as I understood, this problem related to "C.VideoCapture_Open" in videoio.go and required to install "libv4l-dev" (according to https://github.com/opencv/opencv/issues/6157)

Now I do not have a sudo permission to install this library, but I can ask admin to install this.
I just want to confirmed this is the right direction to solve this issue, or should I do something else?

Any tips would be helpful.
Thank you very much.

Best
